src/garage/sampler/on_policy_vectorized_sampler.py

```diff
@@ -153,6 +157,9 @@ def obtain_samples(self, itr, batch_size=None, whole_paths=True):
 
             process_time += time.time() - t
             pbar.inc(len(obses))
+
+            if policy.state_include_action:
```

Comment:
this is a terrible interface. only `Policy` should be mutating `agent_infos`, i.e. to the user of this API, `agent_info` should be a completely opaque data blob.

Reply:
originally I am passing it as `next_action` in`agent_infos` so policy can use that as `prev_action` in next iteration. Would that be better?

Reply:
yes. the policy can put whatever it wants in `agent_infos` and if it needs information from the past, that's where you put it.